### PR TITLE
Install chainer in editable mode

### PR DESCRIPTION
**Pull request by asi1024**
Fixes chainer/chainer#7553.

### CODE REVIEW

**Comment by niboshi** [comment]
How about running from a temporary directory?

File: test_cpu.sh [hunk 1]
```diff
@@ -29,4 +27,9 @@ fi
 
 pytest_opts+=(-m "${pytest_marks[*]}")
 
-python -m pytest "${pytest_opts[@]}" tests/chainer_tests
+TESTS_DIR=${PWD}/chainer/tests/chainer_tests
+
+# Move to temporary directory
```
Comment (niboshi) [inline test_cpu.sh]: ```pushd `mktemp -d` ``` is missing?

Comment (asi1024) [reply test_cpu.sh]: Fixed.

**Comment by niboshi** [comment]
Jenkins, test this please

**Comment by asi1024** [comment]
@niboshi Could you take a look?

**Comment by niboshi** [comment]
The fix is not complete, by trying to see if it works:
Jenkins, test this please

**Comment by asi1024** [comment]
Jenkins, test this please.

**Comment by asi1024** [comment]
Jenkins, test this please.

**Comment by toslunar** [comment]
Jenkins, test this please.

**Comment by asi1024** [comment]
Jenkins, test this please.

**Comment by asi1024** [comment]
Jenkins, test this please.

**Comment by asi1024** [comment]
I gave up running tests from outside of `chainer` directory because it is difficult to measure coverages.

**Comment by asi1024** [comment]
Jenkins, test this please.

**Comment by niboshi** [comment]
There is an error:
>17:10:43 tests/cupy_tests/test_cudnn.py:358: in <module>
>17:10:43     @unittest.skipIf(libcudnn.getVersion() < 7500,
>17:10:43 E   NameError: name 'libcudnn' is not defined

**Comment by kmaehashi** [comment]
The above issue should have resolved in https://github.com/cupy/cupy/pull/2284.

pfnCI, test this please.

**Comment by niboshi** [comment]
LGTM

**Comment by niboshi** [comment]
For record, this is rather a "tentative" fix for https://github.com/chainer/chainer/pull/7518.
We should consider reverting this fix and run tests from outside the chainer directory.

**Pull request merged**